File: Data Structures & Algorithms/binary-tree-right-side-view/submission-4.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if root == None:
            return []

        Q = deque()
        out = []

        Q.append(root)
        while Q:
            qSize = len(Q)
            cur = None
            for _ in range(0, qSize):
                cur = Q.popleft()
                if cur.left:
                    Q.append(cur.left)
                if cur.right:
                    Q.append(cur.right)
            
            out.append(cur.val)
        
        return out

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-4.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        if(root == nullptr)
            return {};
        queue<TreeNode *>Q;
        Q.push(root); 
        vector<vector<int>> out;
        while(!Q.empty())
        {
            out.push_back({});
            int qSize = Q.size();
            for(auto i = 0; i < qSize; ++i)
            {
                TreeNode* cur = Q.front();
                Q.pop();
                if(cur->left)
                    Q.push(cur->left);
                if(cur->right)
                    Q.push(cur->right);
                out.back().push_back(cur->val);
            }
        }
        return out;
    }
};
